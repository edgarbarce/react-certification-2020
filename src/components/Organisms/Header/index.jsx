import React from 'react';
import styled from 'styled-components';
import NavbarLeft from '../../Molecules/NavbarLeft';
import NavbarRight from '../../Molecules/NavbarRight';

const Navbar = styled.nav`
  min-height: 70px;
  background: black;
  display: flex;
  min-width: 100%;
  align-items: center;
  justify-content: space-between;
  box-shadow: -2px 5px 4px 0 rgb(0 0 0 / 71%);
`;

function Header() {
  return (
    <Navbar>
      <NavbarLeft />
      <NavbarRight />
    </Navbar>
  );
}

export default Header;
