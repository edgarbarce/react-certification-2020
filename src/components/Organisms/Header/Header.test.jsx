import React from 'react';
import ReactDOM from 'react-dom';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Header from './index';

describe('Test Header Component', () => {
  afterEach(cleanup);
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header />, div);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
