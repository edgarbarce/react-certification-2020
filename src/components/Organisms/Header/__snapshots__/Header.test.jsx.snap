// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header Component matches snapshot 1`] = `
<nav
  className="sc-pFZIQ cLaYzh"
>
  <div
    className="sc-dlfnbm CZBcd"
  >
    <img
      alt="Menu Icon"
      className="sc-gsTCUz byyktf"
      src="/menu.svg"
    />
    <input
      className="sc-bdfBwQ doNcbL"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="sc-fubCfw eKHZNB"
  >
    <label
      className="sc-jSgupP dLsMOc"
    >
      <input
        className="sc-eCssSg cOQNoX"
        defaultChecked={false}
        type="checkbox"
      />
      <span
        className="sc-hKgILt fnAHGc"
        onClick={[Function]}
      />
    </label>
    <label
      className="sc-iBPRYJ kABctc"
      data-testid="labelTest"
    >
      Dark Mode
    </label>
    <img
      className="sc-gKsewC javJma"
      src="/profile.svg"
    />
  </div>
</nav>
`;
